### PR TITLE
Harrison/pinecone backwards compat

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

**Review by dev2049 — APPROVED** [review]
just a naming nit / question

File: langchain/vectorstores/pinecone.py [hunk 1]
```diff
@@ -26,13 +27,13 @@ class Pinecone(VectorStore):
...
             embeddings = OpenAIEmbeddings()
-            vectorstore = Pinecone(index, embeddings.embed_query, "text")
+            vectorstore = Pinecone(index, embeddings, "text")
     """
 
     def __init__(
         self,
         index: Any,
-        embedding_function: Callable,
+        embeddings: Union[Embeddings, Callable],
```
[the remark this answers is not in the document]
Comment (olivier-lacroix) [reply langchain/vectorstores/pinecone.py:36]: I named it that way in the original PR as the intent is to pass it an instance of an `Embeddings` class

**Comment by olivier-lacroix** [comment]
@hwchase17 , any chance to get this one merged in?

**Comment by olivier-lacroix** [comment]
@hwchase17 , is there anything preventing merging this one? Happy to make any change you would require (like changing to the embedding arg instead of embeddings)

**Comment by baskaryan** [comment]
stale, closing

**Comment by olivier-lacroix** [comment]
@baskaryan , I think getting batch embedding in would still be useful. Let me know what needs to be done to get it merged. What needs to be done to get it merged please?

**Comment by baskaryan** [comment]
> @baskaryan , I think getting batch embedding in would still be useful. Let me know what needs to be done to get it merged. What needs to be done to get it merged please?

let me know what you think of #8982

**Pull request closed without merging**